### PR TITLE
Exams by Daniil Timofeev

### PR DESCRIPTION
**Pull request by SharpDevOps10**
I made the general improvement  of the following solutions (Self-esteem - 95)

### CODE REVIEW

**Review by basarabst — APPROVED** [review]
No problems were found. Approved.

File: Tasks/intersection.js [hunk 1]
```diff
@@ -1,25 +1,14 @@
...
-      delete object_1[attribute_name];
-      {
-      }
+'use strict';
+const intersection = (object1, object2) => {
+  const object1Keys = Object.keys(object1);
+  const result = {};
+  for (const key of object1Keys) {
+    if (object2[key] === object1[key]) {
+      result[key] = object1[key];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:8]: You can save `object[key]` to intermediate variable not to repeat execution of the same expression

File: Tasks/plane.js [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
-require('../Tests/plane.js')(const_plane);
+'use strict';
+const planedArray = (arr) => {
+  const res = [];
+  for (const value of arr) {
+    if (Array.isArray(value)) {
+      res.push(...planedArray(value));
+    } else {
+      res.push(value);
+    }
```
Comment (tshemsedinov) [inline Tasks/plane.js:10]: Alternatively you can use ternary here, but it's ok

File: Tasks/plane.js [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
+  }
+  return res;
+};
+const result = planedArray([[1, [[2]], [3, 4], [5], [6, [7, 8]]]]);
+console.log(result);
```
Comment (tshemsedinov) [inline Tasks/plane.js:15]: Hint: Do not need tests/example here, we call tests from different place

**Review by tshemsedinov — APPROVED** [review]

File: Tasks/intersection.js (hunk 1 above)
Comment (SharpDevOps10) [reply Tasks/intersection.js:8]: Thank you for finding my code`s disadvantages. I`ve got it, thus I`ve made new remastered commits with the following fixes. There is the link : https://github.com/HowProgrammingWorks/Exams/compare/master...SharpDevOps10:Exams:exams
P.S new commits are marked as "remastered " and made on Jan 25, 2023

File: Tasks/plane.js (hunk 2 above)
Comment (SharpDevOps10) [reply Tasks/plane.js:10]: P.P.S I`ve changed 'else-if' block into the ternary operator as well. This advice was alternative, hence I`ve marked it as 'optional'

**Pull request closed without merging**